qt/scientific_interfaces/Indirect/Reduction/ISISEnergyTransferModel.cpp

```diff
@@ -0,0 +1,414 @@
+  const unsigned int nSpectra = 1 + spectraMax - spectraMin;
+  return createDetectorGroupingString(static_cast<std::size_t>(nSpectra), static_cast<std::size_t>(nGroups),
+                                      static_cast<std::size_t>(spectraMin));
+}
+
+std::vector<std::string> IETModel::validatePlotData(IETPlotData plotParams) {
+  std::vector<std::string> errors;
+
+  const std::string inputFiles = plotParams.getInputData().getInputFiles();
+  if (inputFiles.empty()) {
+    errors.push_back("You must select a run file.");
+  }
+
+  IETDataValidator validator;
+
+  bool isRunFileValid = !inputFiles.empty();
+  std::string firstFileName = inputFiles.substr(0, inputFiles.find(','));
+
+  std::string conversionError = validator.validateConversionData(plotParams.getConversionData());
+  if (!conversionError.empty()) {
+    errors.push_back(conversionError);
+  }
+
+  std::vector<std::string> backgroundErrors = validator.validateBackgroundData(
+      plotParams.getBackgroundData(), plotParams.getConversionData(), firstFileName, isRunFileValid);
+  errors.insert(errors.end(), backgroundErrors.begin(), backgroundErrors.end());
+
+  errors.erase(std::remove(errors.begin(), errors.end(), ""), errors.end());
+
+  return errors;
+}
+
+void IETModel::plotRawFile(MantidQt::API::BatchAlgorithmRunner *batchAlgoRunner, InstrumentData instData,
+                           IETPlotData plotParams) {
+  using Mantid::specnum_t;
+
+  const std::string inputFiles = plotParams.getInputData().getInputFiles();
+
+  int spectraMin = plotParams.getConversionData().getSpectraMin();
+  int spectraMax = plotParams.getConversionData().getSpectraMax();
+
+  std::string rawFile = inputFiles.substr(0, inputFiles.find(',')); // getting the name of the first file
+  std::filesystem::path rawFileInfo(rawFile);
+  std::string name = rawFileInfo.filename().string();
+
+  auto loadAlg = loadAlgorithm(rawFile, name);
+  if (instData.getInstrument() != "TOSCA") {
+    if (loadAlg->existsProperty("LoadLogFiles")) {
+      loadAlg->setProperty("LoadLogFiles", false);
+    }
+    loadAlg->setPropertyValue("SpectrumMin", std::to_string(spectraMin));
+    loadAlg->setPropertyValue("SpectrumMax", std::to_string(spectraMax));
+  }
+  loadAlg->execute();
+
+  // Rebin the workspace to its self to ensure constant binning
+  auto inputToRebin = std::make_unique<Mantid::API::AlgorithmRuntimeProps>();
+  inputToRebin->setPropertyValue("WorkspaceToMatch", name);
+  inputToRebin->setPropertyValue("WorkspaceToRebin", name);
+  inputToRebin->setPropertyValue("OutputWorkspace", name);
+
+  IAlgorithm_sptr rebinAlg = AlgorithmManager::Instance().create("RebinToWorkspace");
+  rebinAlg->initialize();
+  batchAlgoRunner->addAlgorithm(rebinAlg, std::move(inputToRebin));
+
+  auto inputFromRebin = std::make_unique<Mantid::API::AlgorithmRuntimeProps>();
+  inputFromRebin->setPropertyValue("InputWorkspace", name);
```

Comment:
Following on from @AnthonyLim23 comment about `inputToRebin` - it looks to me like this is just a way of passing the workspace name of the input to rebin to other algorithms, this is already sorted in a variable `name`, so I think it is probably clearer to set the property on the algorithm directly. 
Does  ` batchAlgoRunner->addAlgorithm` needs to have `std::make_unique<Mantid::API::AlgorithmRuntimeProps>();` passed to it, or can you just give it the algorithm instance?

Reply:
Yes, you can give it the algorithm instance directly. As discussed above, I will remove the whole algorithm